clang/test/SemaSYCL/intel-fpga-local-ast.cpp

```diff
@@ -0,0 +1,537 @@
+  [[intel::numbanks(4), intel::bankwidth(16)]] int var_numbanks_bankwidth;
+  [[intel::bank_bits(2, 3), intel::bankwidth(16)]] int var_bank_bits_width;
+  [[intel::max_replicates(2)]] int var_max_repl;
+  [[intel::simple_dual_port]] int var_dual_port;
+  [[intel::force_pow2_depth(1)]] int var_force_p2d;
+  [[intel::force_pow2_depth(1)]] const int const_force_p2d[64] = {0, 1};
+
+  // Check duplicate argument values with implicit memory attribute.
+  //CHECK: VarDecl{{.*}}var_max_replicates
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelMaxReplicatesAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}12
+  //CHECK-NEXT: IntegerLiteral{{.*}}12{{$}}
+  [[intel::max_replicates(12)]]
+  [[intel::max_replicates(12)]] int var_max_replicates; // OK
+
+  // Check duplicate argument values.
+  //CHECK: VarDecl{{.*}}var_private_copies
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelPrivateCopiesAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}12
+  //CHECK-NEXT: IntegerLiteral{{.*}}12{{$}}
+  [[intel::private_copies(12)]]
+  [[intel::private_copies(12)]] int var_private_copies; // OK
+
+  // Checking of duplicate argument values.
+  //CHECK: VarDecl{{.*}}var_forcep2d
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit
+  //CHECK: SYCLIntelForcePow2DepthAttr
+  //CHECK-NEXT: ConstantExpr
+  //CHECK-NEXT: value:{{.*}}1
+  //CHECK-NEXT: IntegerLiteral{{.*}}1{{$}}
+  [[intel::force_pow2_depth(1)]]
+  [[intel::force_pow2_depth(1)]] int var_forcep2d; // OK
+
+  // Test for Intel FPGA bankwidth memory attribute duplication.
+  // No diagnostic is emitted because the arguments match.
+  // Duplicate attribute is silently ignored.
+  //CHECK: VarDecl{{.*}}var_bankwidth 'int'
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit Default
+  //CHECK: SYCLIntelBankWidthAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 16
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 16
+  [[intel::bankwidth(16)]]
+  [[intel::bankwidth(16)]] int var_bankwidth; // OK
+
+  // Test for Intel FPGA numbanks memory attribute duplication.
+  // No diagnostic is emitted because the arguments match.
+  // Duplicate attribute is silently ignored.
+  //CHECK: VarDecl{{.*}}var_numbanks 'int'
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit Default
+  //CHECK: SYCLIntelNumBanksAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 8
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 8
+  [[intel::numbanks(8)]]
+  [[intel::numbanks(8)]] int var_numbanks; // OK
+
+  // Checking of different argument values.
+  //CHECK: VarDecl{{.*}}bw_bw 'unsigned int[64]'
+  //CHECK: SYCLIntelMemoryAttr{{.*}}Implicit Default
+  //CHECK: SYCLIntelBankWidthAttr
+  //CHECK-NEXT: ConstantExpr{{.*}}'int'
+  //CHECK-NEXT: value: Int 8
+  //CHECK-NEXT: IntegerLiteral{{.*}}'int' 8
+  //expected-warning@+2{{attribute 'bankwidth' is already applied}}
+  [[intel::bankwidth(8)]] // expected-note {{previous attribute is here}}
```

Comment:
I don't think we need to check diagnostics here since this is an AST test. Diagnostics should be tested in a separate file. 

AST can include a CHECK-NOT to see that the conflicting value is ignored

Reply:
Done